### PR TITLE
Support Infinity Reranker (custom reranking models)

### PR DESCRIPTION
**Pull request by haoshan98**
## Title

Support the calling of reranking models served via Infinity engine.

## Relevant issues

[Feature] https://github.com/BerriAI/litellm/issues/7246

## Type

🆕 New Feature

## Changes

Added `InfinityRerank` class

Testing Script:

```
from litellm import rerank

query = "What is the capital of the United States?"
documents = [
    "Carson City is the capital city of the American state of Nevada.",
    "The Commonwealth of the Northern Mariana Islands is a group of islands in the Pacific Ocean. Its capital is Saipan.",
    "Washington, D.C. is the capital of the United States.",
    "Capital punishment has existed in the United States since before it was a country.",
]

response = rerank(
    model="infinity/BAAI/bge-reranker-v2-m3",  # reranking model served via infinity engine
    query=query,
    documents=documents,
    custom_llm_provider="infinity",
    api_base="<INFINITY_API_BASE>",
    api_key="<INFINITY_API_KEY>",
)
print(response)
```

### CODE REVIEW

**Comment by ghost** [comment]
@ishaan-jaff can you review this PR - since you're working on new model/provider onboarding

File: litellm/llms/infinity/rerank.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+)
+from litellm.types.rerank import (
+    RerankRequest, 
+    RerankResponse, 
+    RerankResponseMeta, 
+    RerankTokens,
+)
+
+
+class InfinityRerank(BaseLLM):
```
Comment (ghost) [inline litellm/llms/infinity/rerank.py]: missing a transformation.py to show how the initial request is translated for the rerank context

Comment (haoshan98) [reply litellm/llms/infinity/rerank.py]: Alright, included

**Comment by ghost** [comment]
@haoshan98 looks good. can you please add a unit test against a mock endpoint in here inside tests/llm_translation_testing/

see jina ai testing for reference - https://github.com/BerriAI/litellm/blob/0fe8bfe87ac9dcc53efb726d28de2853a7038270/tests/llm_translation/test_jina_ai.py#L16

**Comment by ishaan-jaff** [comment]
merging into my base branch - will take over from here @haoshan98 

thanks for the work !

**Pull request merged**